### PR TITLE
Update discovery.kubernetes.md

### PR DESCRIPTION
**Pull request by dhalperin**
Added selectors section to the Limited Searched Namespaces section and changed the description of the section to add the specific labels aspect.

#### PR Checklist

- [ ] CHANGELOG updated
- [ ] Documentation added
- [ ] Tests updated

### CODE REVIEW

File: docs/sources/flow/reference/components/discovery.kubernetes.md
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/discovery.kubernetes.md:416]:
```suggestion
This example limits the searched namespaces and only selects pods with a specific label value attached to them.:
```

**Comment by clayton-cornell** [comment]
@grafana/grafana-agent-maintainers Can someone from the team double check the River example and approve.

File: docs/sources/flow/reference/components/discovery.kubernetes.md
Comment (spartan0x117) [inline docs/sources/flow/reference/components/discovery.kubernetes.md:425]: Looks like some unicode characters that River doesn't like got in here:
```suggestion
  selectors {
    role = "pod"
    label = "app.kubernetes.io/name=prometheus-node-exporter"
  }
```

File: docs/sources/flow/reference/components/discovery.kubernetes.md
Comment (spartan0x117) [inline docs/sources/flow/reference/components/discovery.kubernetes.md:432]: nit:
```suggestion
```

**Review by spartan0x117 — COMMENTED** [review]
A few small corrections, but LGTM once those are added.

**Comment by clayton-cornell** [comment]
@dhalperin Thanks for your input here. Can you review and commit the suggestions. I could do it, but it's nice if you can see what has been suggestions before the PR disappears into the merge history :-) I'll merge once you've had a chance to see/review things here.

**Comment by dhalperin** [comment]
Edits looked good so committed the changes. Feel free to merge when ready.

File: docs/sources/flow/reference/components/discovery.kubernetes.md
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/discovery.kubernetes.md:416]:
```suggestion
This example limits the searched namespaces and only selects pods with a specific label value attached to them:
```

**Review by clayton-cornell — APPROVED** [review]
Looks good

@dhalperin I'll get this merged and published :-)

**Pull request merged**
